compiler-core/src/type_/expression.rs

```diff
@@ -3358,6 +3361,39 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
         Ok(())
     }
 
+    fn check_single_case(&mut self, clauses: &[UntypedClause]) {
+        if clauses.len() > 1 {
```

Comment:
Does this emit a warning if the case is inexhaustive? It shouldn't.

We also want to emit a warning for non-constructor single clause case expressions.